src/sdk.tsx

```diff
@@ -112,7 +114,7 @@ function preloadFactory(componentName: 'performance'): (options: PreloadOptions<
 function preloadFactory(componentName: 'remoteConfig'): (options: PreloadOptions<App['remoteConfig']>) => Promise<App['remoteConfig']>;
 function preloadFactory(componentName: 'storage'): (options: PreloadOptions<App['storage']>) => Promise<App['storage']>;
 function preloadFactory(componentName: ComponentName) {
-  return (options: PreloadOptions<FirebaseInstanceFactory>) => preload(componentName, options.firebaseApp, options.setup).toPromise();
+  return (options: PreloadOptions<FirebaseInstanceFactory>) => firstValueFrom(preload(componentName, options.firebaseApp, options.setup));
```

Comment:
because [`toPromise` is deprecated](https://rxjs.dev/deprecations/to-promise)

Reply:
note: this doesn't exist in rxjs 6

Reply:
TODO: when we have a major that drops rxjs 6 we can switch, for now we can just use `toPromise`

Reply:
done